### DISCUSSION TITLE
How to subset two distinct clouds of cells based on nFeatures and nCounts

### DISCUSSION THREAD

**Discussion by CampbellClark1** [satijalab/seurat#4175] [Q&A] [answered]
Hi,

I'm still new to single cell analysis, and don't really have the experience to tell whether my quality control checks on my data are ok. In both of the samples we ran, we can see two distinct clouds of cells from the ncount_RNA and nfeature_RNA scatter plot.  Ideally I want to run the umap analysis separately on both clouds of cells to see if there is a difference in cell populations between them or whether they are homogeneous. 

Is it possible to subset the two clouds of cells at this point? Normally I would've thought I could run a linear model and then subset based on residuals above and below the line of best fit. But I'm unable to perform a linear model on a seurat object. Does anyone have any suggestions on how to subset here?

![image](https://user-images.githubusercontent.com/79322618/109779048-d7139e80-7c05-11eb-9c28-45375fee5014.png)

![image](https://user-images.githubusercontent.com/79322618/109782830-9cac0080-7c09-11eb-8419-d5edb96e2570.png)

**Comment by timoast** [comment, marked as answer]
You can subset the cells based on any criteria you want to set, for example:

```r
library(Seurat)

cells.keep <- which(pbmc_small$nCount_RNA/4 > pbmc_small$nFeature_RNA)
sub_object <- pbmc_small[, cells.keep]
```

However, I'm not sure if making a separate UMAP for each subset here and comparing them is really the best way to go. I'd suggest processing the data together as one object, and then try to assess whether those cells with higher counts relative to the number of features detected separate into a distinct group.

**Reply by CampbellClark1** [reply]
I ended up adding which cells belonged to which cloud to the meta.data and then analysed from that where I could highlight these cells in the plots. It got me the answers I needed. Thanks so much!
